Add Java and Python examples for Snowpipe Streaming SDK

sfc-gh-aniagarwal opened the pull request:
- Add complete Java Maven project with example code
- Add complete Python project with example code
- Include setup instructions, authentication templates, and .gitignore files
- Update main README with links to both examples
- Use com.snowflake:snowpipe-streaming SDK version 1.0.1

sfc-gh-aniagarwal committed:
Add Java and Python examples for Snowpipe Streaming SDK

- Add complete Java Maven project with example code
- Add complete Python project with example code
- Include setup instructions, authentication templates, and .gitignore files
- Update main README with links to both examples
- Use com.snowflake:snowpipe-streaming SDK version 1.0.1

sfc-gh-chathomas commented on python-example/profile.json.example:
```diff
@@ -0,0 +1,8 @@
+  "user": "your_username",
+  "private_key": "path_to_private_key.pem",
+  "role": "your_role",
+  "authenticator": "snowflake_jwt"
```
We dont need this right?  I havent seen this.  Also we need account.  URL should be account_locator for clarity

sfc-gh-chathomas commented on python-example/requirements.txt:
```diff
@@ -0,0 +1,6 @@
+snowflake-ingest>=2.0.0
+
+# Additional dependencies (usually installed with snowflake-ingest)
+snowflake-connector-python>=3.0.0
```
Do we need this?  I dont use it in my demo

sfc-gh-alhuang replied:
+1. I think this should be optional unless our demo code need this.

sfc-gh-chathomas commented on java-example/profile.json.example:
```diff
@@ -0,0 +1,8 @@
+  "user": "your_username",
+  "private_key": "path_to_private_key.pem",
+  "role": "your_role",
+  "authenticator": "snowflake_jwt"
```
Same comment as python profile.json.  See my comment there

sfc-gh-chathomas reviewed (commented):
Left some comments on the profile.jsons.  Why are we adding the JWT auth config there?  Also we dont have account as a main item.  Also URL should be account_locator I believe to be the most clear

sfc-gh-alhuang commented on python-example/README.md:
```diff
@@ -0,0 +1,134 @@
+
+## Prerequisites
+
+- Python 3.8 or higher
```
Should be Python 3.9+

sfc-gh-alhuang commented on python-example/requirements.txt:
```diff
@@ -0,0 +1,6 @@
+# Snowflake Streaming Ingest SDK
+snowflake-ingest>=2.0.0
```
The package name for ssv2 SDK should be [snowpipe-streaming](https://pypi.org/project/snowpipe-streaming/). Also I think we can ignore the version here and just let pip pick up the latest.

sfc-gh-alhuang commented on python-example/streaming_ingest_example.py:
```diff
@@ -0,0 +1,87 @@
+    
+    print("All rows submitted. Waiting for ingestion to complete...")
+    
+    # Wait for ingestion to complete
```
FYI there's an `waitForCommit` api which do similar logic. I don't have a strong preference here though.

sfc-gh-alhuang commented on python-example/README.md:
```diff
@@ -0,0 +1,134 @@
+## Additional Resources
+
+- [Snowflake Streaming Ingest SDK Documentation](https://docs.snowflake.com/en/user-guide/data-load-snowpipe-streaming)
+- [Snowflake Ingest Python SDK on PyPI](https://pypi.org/project/snowflake-ingest/)
```
https://pypi.org/project/snowpipe-streaming/

sfc-gh-alhuang commented on python-example/README.md:
```diff
@@ -0,0 +1,134 @@
+
+## Additional Resources
+
+- [Snowflake Streaming Ingest SDK Documentation](https://docs.snowflake.com/en/user-guide/data-load-snowpipe-streaming)
```
https://docs.snowflake.com/en/user-guide/snowpipe-streaming/snowpipe-streaming-high-performance-overview

sfc-gh-alhuang reviewed (approved):
Thanks for the work! Left some comment for Python part. Also it might be easier to copy the tutorial [here](https://docs.snowflake.com/en/user-guide/snowpipe-streaming/snowpipe-streaming-high-performance-getting-started).

sfc-gh-aniagarwal committed:
refactor: update SDK examples based on PR feedback

- Change Python package from snowflake-ingest to snowpipe-streaming
- Remove unnecessary snowflake-connector-python dependency
- Update profile.json: add account field, remove authenticator, update URL format
- Replace manual polling with waitForCommit API in both Python and Java examples
- Update READMEs to reflect waitForCommit usage and new profile format
- Remove temporary version check files

sfc-gh-aniagarwal committed:
refactor: use try-with-resources and context managers for resource management

- Java: Use try-with-resources for AutoCloseable client and channel
- Python: Use 'with' statements for context manager support
- Removes manual close() calls in favor of automatic resource cleanup
- Follows SDK best practices as seen in SDK test code

sfc-gh-aniagarwal committed:
revert: use manual polling instead of waitForCommit API

- Revert to manual polling loop to make errors more visible
- Keep try-with-resources (Java) and context managers (Python) for proper resource cleanup
- Update READMEs to reflect manual polling approach
- Manual polling provides better error visibility and debugging

sfc-gh-aniagarwal committed:
fix: use getChannelName() instead of getName() in Java example

- Fix compilation error: channel.getName() does not exist
- Use correct API method: channel.getChannelName()
- Verified with SDK source code and local testing
